### ISSUE TITLE
Use hostname instead of external IP in freeswitch config

### ISSUE THREAD

**Issue by jowlo** [bigbluebutton/bbb-install#285]
Hi there,

i am running BBB on AWS with two instances for a webinar setting. We plan on using one larger instance (`c5.4xlarge`) for BBB and a smaller one (`m5a.large`) for the coturn server.

I have setup DNS records on my domain to point to the instance's public IPs with a TTL of 60. I plan on starting the instances only when needed.

On AWS ( if not using Elastic IPs) the IP address changes every time you stop and start the instance. But with updating the DNS records, this should not be an issue. However, the install script writes the external IP at install time to the FreeSwitch config at `/opt/freeswitch/etc/freeswitch/vars.xml`. If the exernal IP changes, BBB reports a `ICE error 1007` when trying to connect the audio session.

I changed the FreeSwitch config to use a host based entry for `external_rtp_ip` and `external_sip_ip` as given in the examples in the config file. This works well.

Are there any reasons to prefer the literal IP approach the script uses?

**Comment by jowlo** [comment]
I just found https://github.com/bigbluebutton/bbb-install/issues/283#issuecomment-710674363 mentioning `bbb-conf --setip <hostname>`. Would that also resolve the issue?

**Comment by ffdixon** [comment]
Interesting.  We just checked the FreeSWITCH docs for [external_rtp_ip](https://freeswitch.org/confluence/display/FREESWITCH/Sofia+Configuration+Files#SofiaConfigurationFiles-ext-rtp-ip) and it does list an option to use the hostname as `<param name="ext-rtp-ip" value="host:mypublicIP.dyndns.org"/>`.

Not sure if this was recently added, but we always assumed it to be an IP address.

Can you try using the hostname for `external_rtp_ip` and `external_sip_ip` and let us know if it works for your use case.

**Comment by jowlo** [comment]
Yes, that's what I am using at the moment, works fine across reboots with IP changes so far.

**Comment by koliosx** [comment]
Dear Fred.
I've installed bbb using the automated script for external ip's ... So far i've got it working but only on external lan and not to my internal. My real question has to do with the public ip. When it changes, it is not working. Is it one time fix for that ? e.g establishing a dns account reflecting to the real time ip each time? or i have to run the installation script from the beggining?
